airbyte-container-orchestrator/src/main/java/io/airbyte/container_orchestrator/Application.java

```diff
@@ -0,0 +1,89 @@
+import io.micronaut.runtime.Micronaut;
+import jakarta.inject.Named;
+import jakarta.inject.Singleton;
+import java.lang.invoke.MethodHandles;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Entrypoint for the application responsible for launching containers and handling all message
+ * passing for replication, normalization, and dbt. Also, the current version relies on a heartbeat
+ * from a Temporal worker. This will also be removed in the future so this can run fully async.
+ * <p>
+ * This application retrieves most of its configuration from copied files from the calling Temporal
+ * worker.
+ * <p>
+ * This app uses default logging which is directly captured by the calling Temporal worker. In the
+ * future this will need to independently interact with cloud storage.
+ */
+@SuppressWarnings({"PMD.AvoidCatchingThrowable", "PMD.DoNotTerminateVM", "PMD.AvoidFieldNameMatchingTypeName"})
+@Singleton
+public class Application {
+
+  public static void main(final String[] args) {
+    // To mimic previous behavior, assume an exit code of 1 unless Application.run returns otherwise.
+    var exitCode = 1;
+    try (final var ctx = Micronaut.run(Application.class, args)) {
+      exitCode = ctx.getBean(Application.class).run();
+    } catch (final Throwable t) {
+      log.error("could not run {}", t.getMessage(), t);
+    } finally {
+      // this mimics the pre-micronaut code, unsure if there is a better way in micronaut to ensure a
+      // non-zero exit code
+      System.exit(exitCode);
+    }
+  }
+
+  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+  private final String application;
+  private final JobOrchestrator<?> jobOrchestrator;
+  private final AsyncStateManager asyncStateManager;
+
+  public Application(@Named("application") final String application,
+                     final JobOrchestrator<?> jobOrchestrator,
+                     final AsyncStateManager asyncStateManager) {
+    this.application = application;
+    this.jobOrchestrator = jobOrchestrator;
+    this.asyncStateManager = asyncStateManager;
+  }
+
+  /**
+   * Configures logging/mdc scope, and creates all objects necessary to handle state updates.
+   * <p>
+   * Handles state updates (including writing failures) and running the job orchestrator. As much of
+   * the initialization as possible should go in here, so it's logged properly and the state storage
+   * is updated appropriately.
+   */
+  @VisibleForTesting
+  int run() {
+    // set mdc scope for the remaining execution
+    try (final var mdcScope = new MdcScope.Builder()
+        .setLogPrefix(application)
+        .setPrefixColor(LoggingHelper.Color.CYAN_BACKGROUND)
+        .build()) {
+
+      asyncStateManager.write(AsyncKubePodStatus.INITIALIZING);
+      asyncStateManager.write(AsyncKubePodStatus.RUNNING);
+      asyncStateManager.write(AsyncKubePodStatus.SUCCEEDED, jobOrchestrator.runJob().orElse(""));
```

Comment:
to confirm: this is only for tests yes?

Reply:
caught my eye because the logic here seems too simple

Reply:
No.  This is the `business logic` of this code.  

How it _use_ to work:
1. `write(INITIALIZING)`
2. block until the configuration files have been copied over via `kubectl cp`
3. `write(RUNNING)`
4. run job
5 `write(SUCCEEDED)`

How it works now: 
1. init-container blocks until configuration files have been copied over via `kubectl cp`
2. `write(INITIALIZING)` // not really necessary any more, as it immediately changes
3. `write(RUNNING)`
3.  run job
4. `write(SUCCEEDED)`

Reply:
All the setup and configuration code that used to exist and run between `INITIALIZING` and `RUNNING` no longer exists has that is now handled by the `initContainer` and `Micronaut`.

Reply:
I see, thanks for explaining.

feels like `INITIALIZING` should be moved to the async orchestrator pod process. or removed entirely if we don't think it's useful to track that an attempt was even made.

Reply:
nit: 

at least for me, writing `      asyncStateManager.write(AsyncKubePodStatus.SUCCEEDED, jobOrchestrator.runJob().orElse(""));` is somewhat confusing since it's not immediately clear `runJob` is blocking.

Would have preferred a more linear style to clarify that.

Reply:
Agreed, seems like we could remove it entirely.

Reply:
thinking more, I feel like it's useful to keep for debugging. would make it obvious if anything crashed between the pod spinning up and actually running